### PR TITLE
Add size-limit

### PR DESCRIPTION
**Pull request by phryneas**
Let's see what the CI does here.

### Checklist:

- [ ] If this PR contains changes to the library itself (not necessary for e.g. docs updates), please include a changeset (see [CONTRIBUTING.md](https://github.com/apollographql/apollo-client/blob/main/CONTRIBUTING.md#changesets))
- [ ] If this PR is a new feature, please reference an issue where a consensus about the design was reached (not necessary for small changes)
- [ ] Make sure all of the significant new logic is covered by tests

### CODE REVIEW

**Comment by phryneas** [comment]
So right now, the CI check doesn't add a comment, because we don't have the size-limit in the target branch yet - so it cannot run there for a size comparison. Here are the sizes we currently do get as an output:

```js
[
  {
    "name": "dist/apollo-client.min.cjs",
    "passed": true,
    "size": 36616,
    "sizeLimit": 36640
  },
  {
    "name": "import { ApolloClient, InMemoryCache, HttpLink } from \"dist/main.cjs\"",
    "size": 46009
  },
  {
    "name": "import { ApolloClient, InMemoryCache, HttpLink } from \"dist/index.js\"",
    "passed": true,
    "size": 34979,
    "sizeLimit": 34990
  },
  {
    "name": "import { ApolloProvider } from \"dist/react/index.js\"",
    "size": 867
  },
  {
    "name": "import { useQuery } from \"dist/react/index.js\"",
    "size": 26182
  },
  {
    "name": "import { useLazyQuery } from \"dist/react/index.js\"",
    "size": 26527
  },
  {
    "name": "import { useMutation } from \"dist/react/index.js\"",
    "size": 24671
  },
  {
    "name": "import { useSubscription } from \"dist/react/index.js\"",
    "size": 2414
  },
  {
    "name": "import { useSuspenseQuery_experimental } from \"dist/react/index.js\"",
    "size": 3619
  },
  {
    "name": "import { useFragment_experimental } from \"dist/react/index.js\"",
    "size": 1821
  }
]
```

This clearly shows that our `useQuery`, `useLazyQuery` and `useMutation` hooks have imports that pull in the whole `core` bundle and that we should fix that up. 

The necessary change for that is
```diff
-import { mergeOptions } from '../../core';
+import { mergeOptions } from '../../utilities';
```
and I'll do that in a follow-up PR (after this is in the 3.8 branch, should we decide to take it) - it will drop those sizes to <6kb for those hooks.

**Comment by phryneas** [comment]
Oh, and to add that context - if the action would have been successful, it would look like this:

**Review by jerelmiller — COMMENTED** [review]
I'm willing to try it. Thanks a bunch!

File: package.json [hunk 1]
```diff
@@ -105,6 +105,8 @@
     "@changesets/cli": "2.26.1",
     "@graphql-tools/schema": "9.0.19",
     "@rollup/plugin-node-resolve": "11.2.1",
+    "@size-limit/esbuild-why": "^8.2.4",
```
Comment (alessbell) [inline package.json:108]: [nit] dev dependencies are pinned to exact versions (same comment for all added lines here)

Comment (phryneas) [reply package.json]: done :)

**Pull request merged**